src/plcc/load_spec/parse_spec/parse_semantic_spec/parse_codeFragments.py

```diff
@@ -0,0 +1,71 @@
+from dataclasses import dataclass
+from .parse_targetLocator import TargetLocator, parse_targetLocator
+from plcc.load_spec.load_rough_spec.parse_lines import Line
+from plcc.load_spec.load_rough_spec.parse_blocks import Block
+from plcc.load_spec.load_rough_spec.parse_dividers import Divider
+import re
+
+@dataclass
+class CodeFragment:
+    targetLocator: TargetLocator
+    block: Block
+
+def parse_codeFragments(lines_and_blocks: list[Line | Block]):
+    parser = CodeFragmentParser(lines_and_blocks)
+    return parser.parse()
+
+class CodeFragmentParser:
+    def __init__(self, lines_and_blocks: list[Line | Block]):
+        self.lines_and_blocks = lines_and_blocks
+        self.codeFragmentList = []
+        self.targetLocator = None
+        self.targetLocator_regex = r'^(\w+)(?::([a-z]+))?\s*(?:#.*)?$'
+
+    def parse(self):
+        for obj in self.lines_and_blocks:
+
+            handler = {
```

Comment:
Extract another function or two here...

def parse(self):
    for obj in self.lines_and_blocks:
        self._parse_line_or_block(obj)
    ...